### PR TITLE
Content creators implementation 

### PR DESCRIPTION
**Pull request by matthewkol186**
This PR completes the implementation of content creators. Content creators dynamically add new items into the system, and can be extended by the user to have different behaviors (e.g., generating items with attributes that come from a non-Bernoulli distribution).  Each recommendation system model has its own method of "processing" new items; for example, popularity recommender represents each new item with a popularity of 0, while content filtering represents each new item with its exact true underlyiing attributes.

### CODE REVIEW

File: trecs/components/creators.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        chosen_profiles = self.actual_creator_profiles[creator_mask == 1, :]
+        # for each creator that will add new items, generate Bernoulli trial
+        # for item attributes
+        items = Generator(seed=self.seed).binomial(
+            1, chosen_profiles.reshape(-1), chosen_profiles.size
+        )
+        return items.reshape(chosen_profiles.shape)
+
+    def update_profiles(self, interactions, items):
+        """In the case of dynamic user profiles, we update the user's actual
```
Comment (elucherini) [inline trecs/components/creators.py:131]: This should be "dynamic creator profiles", right?

File: trecs/components/users.py [hunk 2]
```diff
@@ -226,6 +226,27 @@ def compute_user_scores(self, item_attributes):
 
         self.store_state()
 
+    def score_new_items(self, new_items):
+        """
+        Computes and stores the actual scores that users assign to any new
+        items that enter the compatible with the system. Note that we
```
Comment (elucherini) [inline trecs/components/users.py:232]: Please fix this line: "that enter the compatible"

File: trecs/components/users.py [hunk 3]
```diff
@@ -226,6 +226,27 @@ def compute_user_scores(self, item_attributes):
...
+        expect that the score_fn attribute to be set to some callable function which takes item
```
Comment (elucherini) [inline trecs/components/users.py:233]: "be set" instead of "to be set"?

File: trecs/models/recommender.py [hunk 4]
```diff
@@ -260,6 +267,7 @@ def update_predicted_scores(self):
         user_profiles = self.users_hat
         item_attributes = self.items_hat
         predicted_scores = self.score_fn(user_profiles, item_attributes)
+        # import pdb; pdb.set_trace()
```
Comment (elucherini) [inline trecs/models/recommender.py:270]: just curious, what was this for? Should it stay?

Comment (matthewkol186) [reply trecs/models/recommender.py:270]: Oops, nope! Got rid of it!

Comment (matthewkol186) [reply trecs/models/recommender.py]: it was for aiding w/ debugging :)

**Pull request merged**